src/qt/anybutton.cpp

```diff
@@ -24,18 +25,87 @@ class wxQtPushButton : public wxQtEventSignalHandler< QPushButton, wxAnyButton >
 {
 
 public:
+    virtual bool eventFilter(QObject *watched, QEvent *event);
     wxQtPushButton( wxWindow *parent, wxAnyButton *handler);
 
 private:
     void clicked(bool);
+    wxAnyButton *m_button;
 };
 
 wxQtPushButton::wxQtPushButton(wxWindow *parent, wxAnyButton *handler)
     : wxQtEventSignalHandler< QPushButton, wxAnyButton >( parent, handler )
 {
+    m_button = handler;
+    installEventFilter( this );
```

Comment:
@catalinr ,
Because event filter needs to be installed in order to filter events. You can't just create an event filter hoping that it will work - it has to be applied.